### PR TITLE
Add Smarts Strings for Chlorobenzene and Benzonitrile

### PR DESCRIPTION
**Pull request by Argon1999**
### PR Summary:
Added Smarts Strings for Chlorobenzene and Benzonitrile, they atom typed correctly. Tested Density and Self Diffusivity got decent results when comparing to literature values.
### PR Checklist
------------
 - [ ] Includes appropriate unit test(s)
 - [ ] Appropriate docstring(s) are added/updated
 - [ ] Code is (approximately) PEP8 compliant
 - [ ] Issue(s) raised/addressed?

### CODE REVIEW

**Comment by ahy3nz** [comment]
Toluene, propanenitrile, o-xylene, 124-trimethylbenzene, 2-methylphenol, 3-methylphenol, 4-methylphenol, and ethyl benzene failed for testopls::test_atomtyping. 

Atomtyping conflicts: 260 vs 145, 261 vs 754, 262 vs 753. Check back at the definitions for those atom types, it's possible your PR's definitions are too generic, or maybe those old atom-types need to be amended

**Comment by mattwthompson** [comment]
Also please add some test molecules in MOL2 format and update the `implemented_opls_tests.txt` file. You can look through existing files in `foyer/opls_validation/` or i.e. https://github.com/mosdef-hub/foyer/pull/278 for reference

**Comment by rmatsum836** [comment]
Thanks for your feedback.  @Argon1999 is currently running some simulations of bulk chlorobenzene and benzonitrile to verify several properties.  Afterwards, we will add some molecules, update `implemented_opls_test.txt`, and check the atom typing conflicts.

**Comment by rmatsum836** [comment]
I think tests will pass now.  Should just need to add the mol2 files and update `implemented_opls_tests.txt` now.

**Comment by rmatsum836** [comment]
Update on this: Benzonitrile is atom typing.  We replaced the gro and top files in the `opls_validation` directory with a mol2 file, per #309.  However we are now getting an error due to missing dihedral types.  This also brings up the question of how we want to handle `assert_dihedral_params` for testing atom types.

One of the dihedral types we're missing is in the benonitrile ring (CA-CA-CA-CA), but we can't find it in the force field files in GROMACS.  Searching other sources to see if we can find the parameters for this dihedral.

**Comment by mattwthompson** [comment]
Normally I'd go to http://virtualchemistry.org/ (the companion to David van der Spoel's big JCTC paper: https://pubs.acs.org/doi/abs/10.1021/ct200731v) since I think many of our files originally came from there, but it seems to be offline ....

File: foyer/forcefields/xml/oplsaa.xml [hunk 1]
```diff
@@ -259,11 +259,11 @@
   <Type name="opls_257" class="opls_257" element="C" mass="12.011"/>
   <Type name="opls_258" class="opls_258" element="C" mass="12.011"/>
   <Type name="opls_259" class="opls_259" element="C" mass="12.011"/>
-  <Type name="opls_260" class="opls_260" element="C" mass="12.011"/>
-  <Type name="opls_261" class="opls_261" element="C" mass="12.011"/>
-  <Type name="opls_262" class="opls_262" element="N" mass="14.0067"/>
-  <Type name="opls_263" class="opls_263" element="C" mass="12.011"/>
-  <Type name="opls_264" class="opls_264" element="Cl" mass="35.453"/>
+  <Type name="opls_260" class="CA" element="C" mass="12.011" def="[C;X3;r6]([C;X2;%opls_261])" overrides = "opls_145" desc="Benzonitrile C(CN)" doi="10.1021/ja9621760"/>
+  <Type name="opls_261" class="CZ" element="C" mass="12.011" def="[C;X2]([C;X3;r6])" overrides = "opls_754" desc="Benzonitrile (benzene)C(N)" doi="10.1021/ja9621760"/>
```
Comment (justinGilmer) [inline foyer/forcefields/xml/oplsaa.xml]: The descriptions seem a bit confusing to me. Could they be re-worded a bit? For example:
```suggestion
  <Type name="opls_261" class="CZ" element="C" mass="12.011" def="[C;X2]([C;X3;r6])" overrides = "opls_754" desc="Benzonitrile: carbon in nitrile group (CN) " doi="10.1021/ja9621760"/>
```

**Comment by rmatsum836** [comment]
Thanks for fixing this @Argon1999.  @mosdef-hub/mosdef-contributors I think this is good to go if I could get another pair of eyes on this.

**Comment by rmatsum836** [comment]
JK, `test_from_mbuild_customtype` is failing for some reason on two of the AZP builds.

**Comment by daico007** [comment]
This fail seems to relate to the new box class, so I am looking into this now

**Comment by rmatsum836** [comment]
Tests are now passing.  @mosdef-hub/mosdef-contributors this should be good to be reviewed now.

File: foyer/forcefields/xml/oplsaa.xml [hunk 2]
```diff
@@ -259,11 +259,11 @@
...
   <Type name="opls_259" class="opls_259" element="C" mass="12.011"/>
-  <Type name="opls_260" class="opls_260" element="C" mass="12.011"/>
-  <Type name="opls_261" class="opls_261" element="C" mass="12.011"/>
-  <Type name="opls_262" class="opls_262" element="N" mass="14.0067"/>
-  <Type name="opls_263" class="opls_263" element="C" mass="12.011"/>
-  <Type name="opls_264" class="opls_264" element="Cl" mass="35.453"/>
+  <Type name="opls_260" class="CA" element="C" mass="12.011" def="[C;X3;r6]([C;X2;%opls_261])" overrides = "opls_145" desc="Benzonitrile C(CN)" doi="10.1021/ja9621760"/>
+  <Type name="opls_261" class="CZ" element="C" mass="12.011" def="[C;X2]([C;X3;r6])" overrides = "opls_754" desc="Benzonitrile: carbon in nitrile group (CN) " doi="10.1021/ja9621760"/>
+  <Type name="opls_262" class="NZ" element="N" mass="14.0067" def="[N;X1][C;%opls_261]" overrides = "opls_753" desc="Benzonitrile Nitrogen(benzene-C)N" doi="10.1021/ja9621760"/>
+  <Type name="opls_263" class="CA" element="C" mass="12.011" def="[C;X3;r6]1(Cl)[C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6]1" overrides="opls_145" desc="Chlorobenzene C(Cl)" doi="10.1021/ja9621760"/>
```
Comment (mattwthompson) [inline foyer/forcefields/xml/oplsaa.xml]:
```suggestion
  <Type name="opls_263" class="CA" element="C" mass="12.011" def="[C;X3;r6](Cl)1[C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6]1" overrides="opls_145" desc="Chlorobenzene C(Cl)" doi="10.1021/ja9621760"/>
```
I am just curious if this works too, no need to make this change. I wonder if the branch can be defined both inside and outside the ring

Comment (rmatsum836) [reply foyer/forcefields/xml/oplsaa.xml]: @Argon1999 could you add gro and top files for chlorobenzene in a `Chlorobenzene` directory in `opls_validation`?

File: foyer/opls_validation/chlorobenzene/chlorobenzene.gro [hunk 3]
```diff
@@ -0,0 +1,2403 @@
+GROningen MAchine for Chemical Simulation
+ 2400
```
Comment (justinGilmer) [inline foyer/opls_validation/chlorobenzene/chlorobenzene.gro]: Can we update this file to just have 1 cholorobenzene? Will make the testing a bit faster.

**Comment by rmatsum836** [comment]
Really went down a rabbit hole with this one.  I _think_ that we're much better off reading `gro` and `top` files compared to reading in a `mol2`.  ParmEd assumes a certain naming convention to grab elements by name, and is unable to read in Chlorine by `CL` or `Cl`.  However, if a ParmEd structure is initialized with a `top` file, then the element is read by mass and chlorine is read in correctly.

What caused us problems in this case is that we had _both_ `top` and `mol2` files.  In `test_atomtyping`, we loop through the files in each directory.  As a result, the ParmEd structure loaded in by a `top` file would get overwritten by a structure loaded in by a `mol2`file.  I think we should change the structure of this code where the for loop is broken if we find a `top` file in the directory.

**Comment by daico007** [comment]
@rmatsum836 I think use `mb.load` instead of `pmd.load` in the unit test for the mol2 case should fix this. We are having the same issues a while ago with this over mbuild, and we decided to enforce molecule's name capitalization before shipping to Parmed to fix that.

**Comment by rmatsum836** [comment]
> @rmatsum836 I think use `mb.load` instead of `pmd.load` in the unit test for the mol2 case should fix this. We are having the same issues a while ago with this over mbuild, and we decided to enforce molecule's name capitalization before shipping to Parmed to fix that.

I thought about that too, but wasn't sure if we wanted to go from `mb.Compound` to `pmd.Structure` for these tests.  However, I don't really see any negatives to this fix.

**Comment by mattwthompson** [comment]
My only (tiny) concern is that we're basing a Foyer test off of a quirk in mBuild. I'm fine preferring MDTraj over ParmEd since it seems like it has the better MOL2 reader. I'll raise an issue there to see if we can clear things up.

**Comment by rmatsum836** [comment]
Regardless, the tests are passing so should be good to merge.  Not sure why there is a drop in coverage though.

**Comment by ahy3nz** [comment]
> > @rmatsum836 I think use `mb.load` instead of `pmd.load` in the unit test for the mol2 case should fix this. We are having the same issues a while ago with this over mbuild, and we decided to enforce molecule's name capitalization before shipping to Parmed to fix that.
> 
> I thought about that too, but wasn't sure if we wanted to go from `mb.Compound` to `pmd.Structure` for these tests. However, I don't really see any negatives to this fix.

The whole reason we use `pmd.load_file` is because the resultant `pmd.Structure.atoms` have both `name`(element) and `type` (atom-type). Going back to mbuild loses any atom type information, which is the ["answer" used to validate the atom-typing](https://github.com/mosdef-hub/foyer/blob/master/foyer/tests/utils.py#L7)

**Pull request merged**